src/pymodaq/utils/parameter/utils.py

```diff
@@ -1,14 +1,40 @@
 from __future__ import annotations
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, List
 import numpy as np
 from collections import OrderedDict
-
+from dataclasses import dataclass
 from pymodaq.utils.daq_utils import find_keys_from_val
 
 if TYPE_CHECKING:
     from pymodaq.utils.parameter import Parameter
 
 
+class ParameterWithPath:
+    """ holds together a Parameter object and its full path
+
+    To be used when communicating between TCPIP to reconstruct properly the Parameter
+
+    Parameters
+    ----------
+    parameter: Parameter
+        a Parameter object
+    path: full path of the parameter, if None it is constructed from the parameter parents
+    """
+    def __init__(self, parameter: Parameter, path: List[str] = None):
+        self._parameter = parameter
+        if path is None:
+            path = get_param_path(parameter)
+        self._path = path
+
+    @property
+    def parameter(self):
+        return self._parameter
+
+    @property
+    def path(self):
+        return self._path
```

Comment:
If you use type hints for input, why not also for output?
```suggestion
    @property
    def parameter(self) -> Parameter:
        return self._parameter

    @property
    def path(self) -> List[str]:
        return self._path
```